### PR TITLE
[WIP] HDP

### PR DESCRIPTION
**Pull request by bhargavvader**
This is to address issues #901, and #952, and go towards fixing #945  - basically to attempt to clean up HDP as much as possible.

This includes only the HDP changes from the closed #996. 

I'll make the other cosmetic changes, and moving the appropriate methods to `utils` and `matutils` in a different PR.

### CODE REVIEW

**Comment by bhargavvader** [comment]
@tmylk , only python 2.6 has failed with `ImportError: cannot import name interfaces`. 
Other tests pass fine.

**Comment by tmylk** [comment]
That test error will be fixed after #1056 merged in

File: gensim/models/hdpmodel.py [hunk 1]
```diff
@@ -56,6 +56,21 @@ def dirichlet_expectation(alpha):
     return(sp.psi(alpha) - sp.psi(np.sum(alpha, 1))[:, np.newaxis])
 
 
+def get_random_state(seed):
```
Comment (tmylk) [inline gensim/models/hdpmodel.py]: why copy-paste from LdaModel? Should it be moved to utils?

Comment (bhargavvader) [inline gensim/models/hdpmodel.py]: I'll move the common `ldamodel` and `hdpmodel` methods to the respective `utils` and `matutils` files after this PR is merged.

**Comment by bhargavvader** [comment]
Cool, the remaining changes I made in the old PR I will address in a new PR. Will wait for #1056 is done then.

File: gensim/models/hdpmodel.py (hunk 1 above)
Comment (bhargavvader) [inline gensim/models/hdpmodel.py]: Need to make a few other changes to `utils` and `matutils` as well

**Comment by bhargavvader** [comment]
@tmylk what else would you want done on this PR?

File: gensim/models/hdpmodel.py (hunk 1 above)
Comment (tmylk) [inline gensim/models/hdpmodel.py]: This should be done as a part of this PR. Duplicate code will not be merged.

File: gensim/models/hdpmodel.py [hunk 2]
```diff
@@ -510,6 +542,16 @@ def hdp_to_lda(self):
 
         return (alpha, beta)
 
+    def suggested_lda_model(self):
+        """
+        Returns closest corresponding ldamodel object corresponding to current hdp model.
+        The num_topics is m_T (default is 150) so as to preserve the matrice shapes when we assign alpha and beta.
```
Comment (tmylk) [inline gensim/models/hdpmodel.py]: how is it different from `hdp_to_lda`? Add a comment

Comment (bhargavvader) [inline gensim/models/hdpmodel.py]: Removed duplicate code, added comment.

**Comment by bhargavvader** [comment]
I've moved the `get_random_state` to utils from `ldamodel` and made them both call it from `utils`.

**Comment by bhargavvader** [comment]
@tmylk I've addressed all your comments.

**Comment by tmylk** [comment]
Thanks for the PR!

File: gensim/utils.py [hunk 3]
```diff
@@ -80,6 +81,21 @@ def smart_open(fname, mode='rb'):
...
+
+         Method originally from maciejkula/glove-python, and written by @joshloyal
+     """
+     if seed is None or seed is numpy.random:
+         return numpy.random.mtrand._rand
+     if isinstance(seed, (numbers.Integral, numpy.integer)):
+         return numpy.random.RandomState(seed)
+     if isinstance(seed, numpy.random.RandomState):
+        return seed
+     raise ValueError('%r cannot be used to seed a numpy.random.RandomState'
```
Comment (piskvorky) [inline gensim/utils.py]: No vertical indent in gensim, please use hanging indent.

@tmylk this keeps happening over and over.

Comment (bhargavvader) [inline gensim/utils.py]: My bad, I had just copy-pasted this from the existing `ldamodel` code and missed this. Fixing it in a new PR where I make some changes to `utils`.

Comment (piskvorky) [inline gensim/utils.py]: @bhargavvader Thanks!

**Pull request merged**